docs/installation.rst

```diff
@@ -332,6 +332,7 @@ Use the Cygwin installer to install the dependencies::
     libssl-devel libxxhash-devel liblz4-devel libzstd-devel
     binutils gcc-g++ git make openssh
 
+Make sure to use a virtual environment to avoid confusions with any Python installed on Windows.
```

Comment:
Hmm, sounds like you ran into a problem without doing this?

But what was that? I'ld assume that if one invokes python on a cygwin prompt, one would get the cygwin-python, right (not the native windows python)?

Reply:
Cygwin is a bit weird here. As Cygwin is kind of a win32 environment, it does not have any problems executing any `.exe` files it rans into. Also, Cygwin seems to just extend the Windows `$PATH` with its own `$PATH`:

```cmd
C:\Users\marck>echo %PATH%
C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files\Java\jdk1.8.0_181\bin;C:\Program Files (x86)\PDFtk\bin\;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Program Files\nodejs\;C:\Program Files\dotnet\;C:\Program Files\GitHub CLI\;C:\Program Files\Git\cmd;C:\Program Files\TortoiseGit\bin;C:\Program Files\PuTTY\;C:\Users\marck\AppData\Local\Programs\Python\Python38-32\Scripts\;C:\Users\marck\AppData\Local\Programs\Python\Python38-32\;C:\Users\marck\AppData\Local\Microsoft\WindowsApps;C:\Users\marck\AppData\Local\Programs\Microsoft VS Code\bin;C:\Program Files\kdiff3;C:\Program Files (x86)\Nmap;C:\Users\marck\_Stuff\Apps\ffmpeg-5.0-full_build\bin;C:\Users\marck\AppData\Roaming\npm;C:\Users\marck\AppData\Local\JetBrains\Toolbox\scripts;C:\Users\marck\.dotnet\tools
```

```sh
$ echo $PATH
/usr/local/bin:/usr/bin:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Program Files (x86)/Common Files/Oracle/Java/javapath:/cygdrive/c/WINDOWS/system32:/cygdrive/c/WINDOWS:/cygdrive/c/WINDOWS/System32/Wbem:/cygdrive/c/WINDOWS/System32/WindowsPowerShell/v1.0:/cygdrive/c/WINDOWS/System32/OpenSSH:/cygdrive/c/ProgramData/chocolatey/bin:/cygdrive/c/Program Files/Java/jdk1.8.0_181/bin:/cygdrive/c/Program Files (x86)/PDFtk/bin:/cygdrive/c/Program Files/Docker/Docker/resources/bin:/cygdrive/c/ProgramData/DockerDesktop/version-bin:/cygdrive/c/Program Files/nodejs:/cygdrive/c/Program Files/dotnet:/cygdrive/c/Program Files/GitHub CLI:/cygdrive/c/Program Files/Git/cmd:/cygdrive/c/Program Files/TortoiseGit/bin:/cygdrive/c/Program Files/PuTTY:/cygdrive/c/Users/marck/AppData/Local/Programs/Python/Python38-32/Scripts:/cygdrive/c/Users/marck/AppData/Local/Programs/Python/Python38-32:/cygdrive/c/Users/marck/AppData/Local/Microsoft/WindowsApps:/cygdrive/c/Users/marck/AppData/Local/Programs/Microsoft VS Code/bin:/cygdrive/c/Program Files/kdiff3:/cygdrive/c/Program Files (x86)/Nmap:/cygdrive/c/Users/marck/_Stuff/Apps/ffmpeg-5.0-full_build/bin:/cygdrive/c/Users/marck/AppData/Roaming/npm:/cygdrive/c/Users/marck/AppData/Local/JetBrains/Toolbox/scripts:/cygdrive/c/Users/marck/.dotnet/tools
```

Therefore Cygwin might pick up the `python` on the Windows `$PATH` and will just run it:

```sh
$ which python
/cygdrive/c/Users/marck/AppData/Local/Programs/Python/Python38-32/python

$ python --version
Python 3.8.2
```

This probably works fine on some basic `*.py` scripts, but might get weird on `pip install`. I actually faced a bunch of `python-3.8`, `python-3.9`, `python3.11` and various `pip` commands - some from my Windows `$PATH` and some from my Cygwin `$PATH`.

Using a virtual environment solved this.

Reply:
OK, thanks for explaining!